File: Hackerrank/Exercise PY/10-TimeConversion.py
import re

def TimeConersion(s):
  timeArr = s.split(":");
  hours = int(timeArr[0]);
  minutes = timeArr[1];
  seconds = re.sub(r'[AP]M', '', timeArr[2]);

  if "PM" in s and hours != 12: hours += 12;
  elif "AM" in s and hours == 12: hours = 0;

  if hours < 10: "0" + hours;

  militaryTime = f"{hours}:{minutes}:{seconds}";
  print(militaryTime)



time = "05:05:45PM";
TimeConersion(time);

File: Hackerrank/Exercise JS/10-TimeConversion.js

function timeConversion(s) { 
  const timeArr = s.split(":");
  let hours = parseInt(timeArr[0], 10);
  const minutes = timeArr[1];
  const seconds = timeArr[2].substring(0,2);
  
  if(s.includes("PM") && hours !== 12){
      hours += 12;
  }else if(s.includes("AM") && hours === 12){
      hours = 0;
  }
  
  hours = String(hours).padStart(2, "0");
  
  const militaryTime = `${hours}:${minutes}:${seconds}`;
  console.log(militaryTime);
}

const time = "07:05:45PM";
timeConversion(time);
